addons/mail/models/res_partner.py

```diff
@@ -261,13 +261,16 @@ def _get_mention_suggestions_domain(self, search):
         ])
 
     @api.model
-    def _search_mention_suggestions(self, domain, limit):
+    def _search_mention_suggestions(self, domain, limit, includeInternalUsers=False):
         domain_is_user = expression.AND([[('user_ids', '!=', False)], [('user_ids.active', '=', True)], domain])
         priority_conditions = [
-            expression.AND([domain_is_user, [('partner_share', '=', False)]]),  # Search partners that are internal users
             domain_is_user,  # Search partners that are users
             domain,  # Search partners that are not users
         ]
+        if includeInternalUsers:
+            priority_conditions.insert(0, expression.AND([[('user_ids', '!=', False)], [('user_ids.active', '=', True)], [('partner_share', '=', False)]]))
+        else:
+            priority_conditions.insert(0, expression.AND([domain_is_user, [('partner_share', '=', False)]]))  # Search partners that are internal users
```

Comment:
Both the domains which are being inserted conditionally are basically same. So, it doesn't make sense to do this conditionally.